router/pkg/config/config.go

```diff
@@ -215,6 +215,28 @@ type AuthorizationConfiguration struct {
 	RejectOperationIfUnauthorized bool `yaml:"reject_operation_if_unauthorized" default:"false" envconfig:"REJECT_OPERATION_IF_UNAUTHORIZED"`
 }
 
+type RateLimitConfiguration struct {
+	Enabled        bool                    `yaml:"enabled" default:"false" envconfig:"RATE_LIMIT_ENABLED"`
+	Strategy       string                  `yaml:"strategy" default:"simple" envconfig:"RATE_LIMIT_STRATEGY" validate:"oneof=simple"`
+	SimpleStrategy RateLimitSimpleStrategy `yaml:"simple_strategy"`
+	Storage        RedisConfiguration      `yaml:"storage"`
+	// Debug ensures that retryAfter and resetAfter are set to stable values for testing
+	Debug bool `yaml:"debug" default:"false" envconfig:"RATE_LIMIT_DEBUG"`
+}
+
+type RedisConfiguration struct {
+	Addr      string `yaml:"addr" default:"localhost:6379" envconfig:"REDIS_ADDR" validate:"required"`
+	Password  string `yaml:"password" envconfig:"REDIS_PASSWORD"`
+	KeyPrefix string `yaml:"key_prefix" default:"cosmo_rate_limit" envconfig:"RATE_LIMIT_REDIS_KEY_PREFIX" validate:"required"`
+}
+
+type RateLimitSimpleStrategy struct {
+	Rate                             int           `yaml:"rate" default:"10" envconfig:"RATE_LIMIT_SIMPLE_RATE" validate:"required,min=1"`
+	Burst                            int           `yaml:"burst" default:"10" envconfig:"RATE_LIMIT_SIMPLE_BURST" validate:"required,min=1"`
+	Period                           time.Duration `yaml:"period" default:"1s" envconfig:"RATE_LIMIT_SIMPLE_PERIOD" validate:"required,min=1s"`
+	RejectExceedingRateLimitRequests bool          `yaml:"reject_exceeding_rate_limit_requests" default:"false" envconfig:"RATE_LIMIT_SIMPLE_REJECT_EXCEEDING_RATE_LIMIT_REQUESTS"`
```

Comment:
I think we can remove "RateLimit" from the name because we are inside the RateLimit section already.

Reply:
Apollo uses the term "reject" for authorization, which I also adopted for authorization and thought I'd stay with the naming. This is an MVP, we can still change a lot of things based on feedback.